### DISCUSSION TITLE
How to work with relationships?

### DISCUSSION THREAD

**Discussion by francoism90** [spatie/laravel-data#220] [Q&A] [answered]
I've implemented `WithData` on my models:
https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/creating-a-data-object#content-quickly-getting-data-from-models-requests-

This is my `PostData` corresponding with my `Post` model, which has a relationship called `user`:
```
class PostData extends Data
{
    public function __construct(
        public int $id,
        public string $name,
        public string $content,
        public UserData $user,
    ) {
    }
}
```

```
class UserData extends Data
{
    public function __construct(
        public int $id,
        public string $name,
        public string $email,
    ) {
    }
}
```

I would expect `Post::firstOrFail($id)->getData();` return a filled `UserData` object - however this isn't the case.

How should you deal with relationships?

Thanks!

**Comment by rubenvanassche** [comment, marked as answer]
At the moment we do not support automatically mapping model relationships onto data objects, in such a case a magic creation method is probably the best solution to the problem!

**Reply by francoism90** [reply]
@rubenvanassche https://spatie.be/docs/laravel-data/v2/as-a-resource/lazy-properties#content-relational-lazy-properties

Exactly what I need! Thanks. :)

**Reply by francoism90** [reply]
@rubenvanassche It would help if you guys could fix the search engine lol.
